### PR TITLE
Explanatory improvements

### PR DESCRIPTION
**Pull request by Steve-A-Orr**
While working through how to set up my environment this last week, I stumbled a lot with a bunch of false assumptions. I've clarified the instructions to negate someone else having my issues and improved the text for readability.

I hope the changes seem reasonable to you.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,128 +4,110 @@
 # The Primo New UI Customization Workflow Development Environment
 
 
-##Structure
+## Structure
 
-- <b>gulp directory</b> : holds the various build scripts for the environment and the configuration file (config.js) in which the target proxy server is defined
+- <b>gulp directory</b> : holds the various build scripts for the environment and the  <b>config.js</b> configuration file in which your target proxy-server must be defined.
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 2]
```diff
@@ -4,128 +4,110 @@
...
 
-- <b>node_modules directory</b> : holds the various 3rd party modules that are required to run the system , Those modules are defined in the:
-`./package.json` file
+- <b>node_modules directory</b> : holds the various third-party modules that are required to run the system. These modules are defined in the <b>package.json</b> file.
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 3]
```diff
@@ -4,128 +4,110 @@
...
-   1. <b> custom </b> : - where you will place your custimization packages
-   2. <b> tmp </b> : just a place to hold some of your temporary files
+   1. <b>custom</b> : - where you will place your customization packages
+   2. <b>tmp</b> : just a place to hold some of your temporary files
 
-##Overview
+## Overview
 
-The development package allows you to configure :
+The development package allows you to configure the following page components (follow the links for details):
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 4]
```diff
@@ -4,128 +4,110 @@
...
-- images
+- [HTML](https://github.com/ExLibrisGroup/primo-explore-package/tree/master/VIEW_CODE/html "html documentation")
 
-- html
+- [Images](https://github.com/ExLibrisGroup/primo-explore-package/tree/master/VIEW_CODE/img "images documentation")
 
-- javascript
+- [JavaScript](https://github.com/ExLibrisGroup/primo-explore-package/tree/master/VIEW_CODE/js "javascript documentation")
 
+For each configuration-type, or for every different [Primo View](https://knowledge.exlibrisgroup.com/Alma/Product_Documentation/Alma_Online_Help_(English)/Alma-Primo_Integration/040Configuring_the_Primo_Front_End_for_an_Alma_Data_Source/010Creating_a_Primo_View), there should be a specified folder named after the View (which adheres to the established directory structure) in the `primo-explore/custom` package folder.
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 5]
```diff
@@ -4,128 +4,110 @@
...
 
-- For each configuration type there is a specified folder in the custom package folder (that can be downloaded from your Primo Back Office, or from the [GitHub Repository](https://github.com/ExLibrisGroup/primo-explore-package "primo-explore-package repository"))
-- In each folder you will find a specific README.md file with recipes/examples.
+This custom View folder can be downloaded from your Primo Back Office, by following `Primo Home > Primo Utilities > UI customization Package Manager`, or started fresh from the [primo-explore-package GitHub repository](https://github.com/ExLibrisGroup/primo-explore-package "primo-explore-package repository"). (The benefit of using this repository is that in each folder you will find a specific README.md file containing recipes and examples.)
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 6]
```diff
@@ -4,128 +4,110 @@
...
 
-- Follow the links below to learn more about the different customization options:
 
-    [CSS](https://github.com/ExLibrisGroup/primo-explore-package/tree/master/VIEW_CODE/css "css documentation")
+## Installation
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 7]
```diff
@@ -4,128 +4,110 @@
...
 
-    [HTML](https://github.com/ExLibrisGroup/primo-explore-package/tree/master/VIEW_CODE/html "html documentation")
+1.  Download the project from this repository and place it on your computer
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 8]
```diff
@@ -4,128 +4,110 @@
...
 
-    [Images](https://github.com/ExLibrisGroup/primo-explore-package/tree/master/VIEW_CODE/img "images documentation")
+2.  Unzip the file you downloaded to a preferred development project folder location
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 9]
```diff
@@ -4,128 +4,110 @@
...
 
-    [JavaScript](https://github.com/ExLibrisGroup/primo-explore-package/tree/master/VIEW_CODE/js "javascript documentation")
+3.  Download and install the [latest LTS version of Node](https://nodejs.org/en/download/)
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 10]
```diff
@@ -4,128 +4,110 @@
...
 
+4.  From command line, run the command : `npm install npm@3.3.12 -g`
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 11]
```diff
@@ -4,128 +4,110 @@
...
 
-##Installation
+5.  Restart your computer
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 12]
```diff
@@ -4,128 +4,110 @@
...
 
--  Download the project from this repository and place it on your pc
+6.  From command line, run the command : `npm install -g gulp`
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 13]
```diff
@@ -4,128 +4,110 @@
...
 
--  Unzip the file you downloaded
+7.  In a <b>new</b> command line window, navigate to the project base directory (`cd \path\to\your\project\folder\primo-explore-devenv`)
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 14]
```diff
@@ -4,128 +4,110 @@
...
--  Restart the computer
-
--  Run `npm install -g gulp` from the command line
-
--  Open a new command line window
-
--  cd to the project base directory (`your-drive-letter:\**\**\primo-explore-devenv`/`your-drive-letter:\**\**\primo-explore-devenv-master`)
-
--  Run `npm install`. This should install all node modules needed for gulp
+8.  From command line, run the command : `npm install` (This should install all node modules needed for gulp.)
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 15]
```diff
@@ -4,128 +4,110 @@
...
 
     ![npm install image](./help_files/npmInstall.png "Running npm install")
 
--  Edit the proxy server (Your Primo Front-End Url) under /gulp/config.js: `var PROXY_SERVER = http://your-server:your-port`,
-
--  Note that for SSL environments (https) define the server as: `var PROXY_SERVER = https://your-server:443`
--  Go to your custom package folder and start your customizations:
-    1.  Download from your Primo Back Office, or from the [GitHub Repository](https://github.com/ExLibrisGroup/primo-explore-package "primo-explore-package repository")) a new package folder.
-    In case you already defined a package and loaded it to the BO - make sure you download it or else you will not see your previous changes.
+9.  Edit Gulp configuration file's <i>proxy server</i> setting, found at <b>gulp/config.js</b> : `var PROXY_SERVER = http://your-server:your-port` (Make sure to use your real Sandbox or Production Primo Front-End URL.) Note that for SSL environments (HTTPS) define the server as: `var PROXY_SERVER = https://your-server:443`
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 16]
```diff
@@ -4,128 +4,110 @@
...
 
-    2.  Save the package folder(the VIEW_CODE folder) into your development environment under the primo-explore/custom directory
+10. Populate your custom View package folder in the custom package folder ("...primo-explore\custom"), by either downloading the view code files from your Primo Back Office or using the [primo-explore-package GitHub repository](https://github.com/ExLibrisGroup/primo-explore-package "primo-explore-package repository")) to start a new package folder. (if you have already defined a view package and loaded it to the BO - make sure you download it or else you will not see, and may overwrite, your previous changes.)
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 17]
```diff
@@ -4,128 +4,110 @@
...
 
-    your directory tree should look similar to this:
+   - If your custom view package folder were to be called "Auto1" then your development environment directory tree should look similar to this:
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 18]
```diff
@@ -4,128 +4,110 @@
...
 
-    2.  Save the package folder(the VIEW_CODE folder) into your development environment under the primo-explore/custom directory
+10. Populate your custom View package folder in the custom package folder ("...primo-explore\custom"), by either downloading the view code files from your Primo Back Office or using the [primo-explore-package GitHub repository](https://github.com/ExLibrisGroup/primo-explore-package "primo-explore-package repository")) to start a new package folder. (if you have already defined a view package and loaded it to the BO - make sure you download it or else you will not see, and may overwrite, your previous changes.)
 
-    your directory tree should look similar to this:
+   - If your custom view package folder were to be called "Auto1" then your development environment directory tree should look similar to this: 
+   ![Directory tree image](./help_files/direcoryTree.png "Directory tree")
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 19]
```diff
@@ -4,128 +4,110 @@
...
+
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 20]
```diff
@@ -4,128 +4,110 @@
...
+10. Populate your custom View package folder in the custom package folder ("...primo-explore\custom"), by either downloading the view code files from your Primo Back Office or using the [primo-explore-package GitHub repository](https://github.com/ExLibrisGroup/primo-explore-package "primo-explore-package repository")) to start a new package folder. (if you have already defined a view package and loaded it to the BO - make sure you download it or else you will not see, and may overwrite, your previous changes.)
 
-    your directory tree should look similar to this:
+   - If your custom view package folder were to be called "Auto1" then your development environment directory tree should look similar to this: 
+   ![Directory tree image](./help_files/direcoryTree.png "Directory tree")
+   
+   - <b>IMPORTANT:</b> The name of your custom view package folder must match an <i>existing</i> view on the proxy server being referenced or the Gulp server will not function properly. For development from scratch, be sure to first create (or copy) a view using the Primo Back Office View Wizard; then you can accomplish your customizations locally using this document.
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 21]
```diff
@@ -4,128 +4,110 @@
...
+
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 22]
```diff
@@ -4,128 +4,110 @@
...
 
-    your directory tree should look similar to this:
+   - If your custom view package folder were to be called "Auto1" then your development environment directory tree should look similar to this: 
+   ![Directory tree image](./help_files/direcoryTree.png "Directory tree")
+   
+   - <b>IMPORTANT:</b> The name of your custom view package folder must match an <i>existing</i> view on the proxy server being referenced or the Gulp server will not function properly. For development from scratch, be sure to first create (or copy) a view using the Primo Back Office View Wizard; then you can accomplish your customizations locally using this document.
+    
+11. Start your code customizations :
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 23]
```diff
@@ -4,128 +4,110 @@
...
-    your directory tree should look similar to this:
+   - If your custom view package folder were to be called "Auto1" then your development environment directory tree should look similar to this: 
+   ![Directory tree image](./help_files/direcoryTree.png "Directory tree")
+   
+   - <b>IMPORTANT:</b> The name of your custom view package folder must match an <i>existing</i> view on the proxy server being referenced or the Gulp server will not function properly. For development from scratch, be sure to first create (or copy) a view using the Primo Back Office View Wizard; then you can accomplish your customizations locally using this document.
+    
+11. Start your code customizations : 
 
-    ![Directory tree image](./help_files/direcoryTree.png "Directory tree")
+   - From command line, go to your custom view package folder : `cd primo-explore\custom\VIEW_CODE`
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 24]
```diff
@@ -4,128 +4,110 @@
...
 
--  In the command line run : `gulp run --view <the VIEW_CODE folder>` - This will start your local server
+   - From command line, run the command : `gulp run --view <the VIEW_CODE folder>` (This will start your local server.)
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 25]
```diff
@@ -4,128 +4,110 @@
...
+
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 26]
```diff
@@ -4,128 +4,110 @@
...
+    
+11. Start your code customizations : 
 
-    ![Directory tree image](./help_files/direcoryTree.png "Directory tree")
+   - From command line, go to your custom view package folder : `cd primo-explore\custom\VIEW_CODE`
 
--  In the command line run : `gulp run --view <the VIEW_CODE folder>` - This will start your local server
+   - From command line, run the command : `gulp run --view <the VIEW_CODE folder>` (This will start your local server.)
+   
+     (For example, running `gulp run --view Auto1` will start the environment taking the customizations from the <b>Auto1</b> folder.)
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 27]
```diff
@@ -4,128 +4,110 @@
...
+
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 28]
```diff
@@ -4,128 +4,110 @@
...
 
-    ![Directory tree image](./help_files/direcoryTree.png "Directory tree")
+   - From command line, go to your custom view package folder : `cd primo-explore\custom\VIEW_CODE`
 
--  In the command line run : `gulp run --view <the VIEW_CODE folder>` - This will start your local server
+   - From command line, run the command : `gulp run --view <the VIEW_CODE folder>` (This will start your local server.)
+   
+     (For example, running `gulp run --view Auto1` will start the environment taking the customizations from the <b>Auto1</b> folder.)
+     
+     ![Server Startup Image](./help_files/serverStartup.png "Server Startup")
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 29]
```diff
@@ -4,128 +4,110 @@
...
 
-    example:
+   - Open a browser and type in the following URL : `localhost:8003/primo-explore/?vid=your-view-code`  (Example: http://localhost:8003/primo-explore/search?vid=Auto1)
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 30]
```diff
@@ -4,128 +4,110 @@
...
 
-    `gulp run --view Auto1`
+   -  Now you should be able to to your customizations with real searches and results, from your previously defined proxy-server. Note: once you start working with this environment, you will discover that the best results are achieved by working in your browser's incognito mode; or you can clear your browser cache before you start the Gulp server.
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 31]
```diff
@@ -4,128 +4,110 @@
...
+
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 32]
```diff
@@ -4,128 +4,110 @@
...
+     
+     ![Server Startup Image](./help_files/serverStartup.png "Server Startup")
 
-    example:
+   - Open a browser and type in the following URL : `localhost:8003/primo-explore/?vid=your-view-code`  (Example: http://localhost:8003/primo-explore/search?vid=Auto1)
 
-    `gulp run --view Auto1`
+   -  Now you should be able to to your customizations with real searches and results, from your previously defined proxy-server. Note: once you start working with this environment, you will discover that the best results are achieved by working in your browser's incognito mode; or you can clear your browser cache before you start the Gulp server.
+   
+   ![Env up Image](./help_files/searchResults.png "Env up")
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 33]
```diff
@@ -4,128 +4,110 @@
...
 
-     will start the environment taking the customizations from the Auto1 folder
+   -  You can get immediate feedback on your code changes by refreshing the browser.
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 34]
```diff
@@ -4,128 +4,110 @@
...
 
-    ![Server Startup Image](./help_files/serverStartup.png "Server Startup")
+   -  Perform your changes according to the documentation/examples in:
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 35]
```diff
@@ -4,128 +4,110 @@
...
 
--  Open a browser and type in the following url:
+      - [CSS](https://github.com/ExLibrisGroup/primo-explore-package/tree/master/VIEW_CODE/css "css documentation")
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 36]
```diff
@@ -4,128 +4,110 @@
...
 
-   `localhost:8003/primo-explore/?vid=your-view-code`
+      - [HTML](https://github.com/ExLibrisGroup/primo-explore-package/tree/master/VIEW_CODE/html "html documentation")
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 37]
```diff
@@ -4,128 +4,110 @@
...
 
-    example: http://localhost:8003/primo-explore/search?vid=Auto1
+      - [Images](https://github.com/ExLibrisGroup/primo-explore-package/tree/master/VIEW_CODE/img "images documentation")
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 38]
```diff
@@ -4,128 +4,110 @@
...
 
--  Now you can perform searches on your browser and receive results from your actual defined proxy server
+      - [JavaScript](https://github.com/ExLibrisGroup/primo-explore-package/tree/master/VIEW_CODE/js "javascript documentation")
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 39]
```diff
@@ -4,128 +4,110 @@
...
 
-    Note: Once you start working with the environment you will discover that the best results are achived by working in incognito mode, or clearing your browser cache before you start the server
 
-    ![Env up Image](./help_files/searchResults.png "Env up")
 
--  You can get immediate feedback on your changes by refreshing the browser
+## Publishing packages
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 40]
```diff
@@ -4,128 +4,110 @@
...
 
--  perform your changes according to the documentation/examples in:
+Once you finish customizing the package, you can zip up that directory and upload it using the Primo BackOffice.
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 41]
```diff
@@ -4,128 +4,110 @@
...
-
-    [JavaScript](https://github.com/ExLibrisGroup/primo-explore-package/tree/master/VIEW_CODE/js "javascript documentation")
-
-
-
-##Publishing packages
--   Once you finish customizing the package , you can zip it and upload it via the Primo BackOffice
--   Run the command (from command line) `gulp create-package`
--   You will be prompted with a menu specifing all of the possible packages you can build:
+1. In a command line window, navigate to the project base directory : `cd \path\to\your\project\folder\primo-explore-devenv`
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 42]
```diff
@@ -4,128 +4,110 @@
...
 
+2. From command line, run the command : `gulp create-package` You will be prompted with a menu specifying all of the possible packages you can build, such as :
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

File: README.md [hunk 43]
```diff
@@ -4,128 +4,110 @@
...
 
     ![Create Package Image](./help_files/createPackage.png "Create Package up")
 
     ![Package Image](./help_files/packages.png "Package up")
 
+3. Log into Primo Back Office and navigate to the <b>UI customization Package manager</b> section : `Primo Home > Primo Utilities > UI customization Package Manager`
```
Comment (noamamit92) [inline README.md]: Explanatory improvements #36

Comment (noamamit92) [inline README.md]: Explanatory improvements #36

**Comment by noamamit92** [comment]
Explanatory improvements #36

File: README.md (hunk 11 above)
Comment (danmichaelo) [reply README.md]: Eeeh.. *what*??

**Pull request merged**